shell/platform/windows/windows_lifecycle_manager.h

```diff
@@ -34,12 +47,43 @@ class WindowsLifecycleManager {
                     std::optional<LPARAM> lparam,
                     UINT exit_code);
 
-  // Intercept top level window messages, only paying attention to WM_CLOSE.
+  // Intercept top level window WM_CLOSE message and listen to events that may
+  // update the application lifecycle.
   bool WindowProc(HWND hwnd, UINT msg, WPARAM w, LPARAM l, LRESULT* result);
```

Comment:
Perhaps consider adding a little more context in comments for outsiders unfamiliar with this codebase:

1. `WindowProc`: for top-level Flutter `HWND`s
2. `ExternalWindowMessage`: for top-level "external" `HWND`s
3. `OnWindowStateEvent`: for "inner" `HWND`s that back Flutter views

Perhaps a different name could help clarify too? Perhaps `OnTopLevelFlutterWindowMessage`, `OnTopLevelExternalWindowMessage`, and `OnWindowStateEvent`? I don't feel strongly about any of this, feel free to use different names or keep the current ones :)